cordova-lib/src/plugman/fetch.js

```diff
@@ -205,8 +206,9 @@ function fetchPlugin(plugin_src, plugins_dir, options) {
 // Helper function for checking expected plugin IDs against reality.
 function checkID(expectedIdAndVersion, pinfo) {
     if (!expectedIdAndVersion) return;
-    var expectedId = expectedIdAndVersion.split('@')[0];
-    var expectedVersion = expectedIdAndVersion.split('@')[1];
+    var parts = cordovaUtil.parseRegistryPluginSpec(expectedIdAndVersion);
+    var expectedId = parts[0] || expectedIdAndVersion;
+    var expectedVersion = parts[1];
     if (expectedId != pinfo.id) {
```

Comment:
Since you use `parseRegistryPluginSpec`, it is possible that `expectedId` would be specified as `@scope/id`, but from what i understand, pluginInfo.id could not contain scope so this comparison will always be truthy in this case